modules/rostests/apitests/kernel32/UEFIFirmware.c

```diff
@@ -0,0 +1,253 @@
+/*
+ * PROJECT:     ReactOS API Tests
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Tests for UEFI Firmware functions
+ * COPYRIGHT:   Copyright 2023 Ratin Gao <ratin@knsoft.org>
+ */
+
+#include "precomp.h"
+
+#include <ndk/psfuncs.h>
+#include <ndk/setypes.h>
+#include <ndk/sefuncs.h>
+#include <ndk/obfuncs.h>
+
+#define _A2W(quote) __A2W(quote)
+#define __A2W(quote) L##quote
```

Comment:
Isn't it already done via `_T` or `TEXT` macro?

Reply:
`#define TEXT(s) L##s` is already defined in `k32.h`, but `_T` or `TEXT` seems depends on ANSI build or Unicode build.
Here convert `"xxx"` to `L"xxx"` explicitly.

Reply:
ah yes correct :+1: